source/runtime.hpp

```diff
@@ -31,14 +31,70 @@ namespace reshade
 	class runtime abstract
 	{
 	public:
+		enum class game
+		{
+			ACO,                // Assassin's Creed Origins
+			PCARS2,				// Project Cars 2
+			PCARS2AVX,			// Project Cars 2 - other exe
+			TITANFALL2,			// TitanFall 2
+			TITANFALL2_TRIAL,   // TitanFall 2 - trial exe
+			SOW,     			// Shadow Of War
+			ELEX,     			// Elex
+			AOM_RELEASE_FINAL,  // Agents Of Mahyem
+			AOM,     			// Agents Of Mahyem - other exe
+			UNKNOWN
+		};
+
+		enum depth_buffer_retrieval_mode {
+			POST_PROCESS = 0,
+			BEFORE_CLEARING_STAGE = 1,
+			AT_OM_STAGE = 2
+		};
+
+		enum depth_buffer_texture_type {
+			BOTH = 0,
+			DEPTH_BUFFER = 1,
+			STENCIL_BUFFER = 2
+		};
+
+		enum class allowed_technique
+		{
+			DISPLAY_DEPTH,
+			SUPER_DEPTH_3D,
+			POLYNOMIAL_BARREL_DISTORTION_M,
+			POLYNOMIAL_BARREL_DISTORTION_S,
+			MXAO
+		};
+
 		/// <summary>
 		/// File path to the current module.
 		/// </summary>
 		static filesystem::path s_reshade_dll_path;
 		/// <summary>
+		/// screen width
+		/// </summary>
+		static unsigned int screen_width;
```

Comment:
Apart from the fact that `static` variables are a bad idea here again, because applications can technically have multiple DX devices / OGL contexts at the same time, which would instantiate the ReShade runtime multiple times, so that this variable would be read/written to by different runtime instances with different size values, there is also no need for these. Instead use `runtime::frame_width()` and `runtime::frame_height()`. You can get access to all the runtime instances in D3D11 via the `_runtimes` field in a `D3D11Device`. Make sure they are all represented, since again, multiple could technically exist (even though this is rare).

Reply:
Ok, but if there are multiple runtimes, which one can i use to retrieve the screen width and screen height in a specific d3d11_context ?

Reply:
I suppose you can just use the first in the array, since at that point you don't know which runtime the game will present on later on.

Reply:
OK, made the change as suggested